### PR TITLE
Require Manual Approval for: unknown hosts + unseen hosts

### PR DESCRIPTION
**Pull request by agrinman**
notable changes:
- ability to disable requiring manual for unknown hosts.
- introduce VerifiedHostAuth type on SignRequests: only gets initialized if the HostAuth is signed correctly. VerifiedHostAuth used everywhere, HostAuth type only used in construction of SignRequests.
- fixed delete known host save bug.

### CODE REVIEW

File: Kryptonite/Policy.swift [hunk 1]
```diff
@@ -24,7 +24,8 @@ class Policy {
         case userApproval = "policy_user_approval"
         case userLastApproved = "policy_user_last_approved"
         case userApprovalInterval = "policy_user_approval_interval"
-        
+        case manualUnknownHostApprovals = "policy_manual_host_approvals"
```
Comment (kcking) [inline Kryptonite/Policy.swift]: minor: should this be policy_manual_unknown_host_approvals?

File: Kryptonite/Request.swift [hunk 2]
```diff
@@ -85,27 +85,25 @@ struct HostAuthVerificationFailed:Error{}
...
 
-        if let potentialHostAuth = hostAuth{
-            guard try potentialHostAuth.verify(sessionID: session) == true
-            else {
-                throw InvalidHostAuthSignature()
-            }
-
-            self.hostAuth = potentialHostAuth
+        if let potentialHostAuth = hostAuth {
+            self.verifiedHostAuth = try? VerifiedHostAuth(session: session, hostAuth: potentialHostAuth)
```
Comment (kcking) [inline Kryptonite/Request.swift]: can we add an explanation here that agent-forwarding and _some_ legitimate host key types will show up as unknown host, but we want to phase it out asap?

File: Kryptonite/Request.swift [hunk 3]
```diff
@@ -115,18 +113,8 @@ struct SignRequest:Jsonable {
...
-            else {
-                throw InvalidHostAuthSignature()
-            }
-
-            self.hostAuth = potentialHostAuth
-        } catch {
-            log("host auth error: \(error)")
-            hostAuth = nil
+        if let potentialHostAuth = try? HostAuth(json: json ~> "host_auth") {
+            verifiedHostAuth = try? VerifiedHostAuth(session: session, hostAuth: potentialHostAuth)
```
Comment (kcking) [inline Kryptonite/Request.swift]: is there an easy way to have the JSON initializer call init so the same hostAuth logic is used for each?

File: Kryptonite/Notify.swift [hunk 4]
```diff
@@ -182,7 +186,7 @@ class Notify {
             content.body = noteBody
             content.sound = UNNotificationSound.default()
             
-            let request = UNNotificationRequest(identifier: "\(message.hash)", content: content, trigger: nil)
+            let request = UNNotificationRequest(identifier: "\(session.id)_\(message.hash)", content: content, trigger: nil)
```
Comment (agrinman) [inline Kryptonite/Notify.swift]: yes

**Pull request merged**